modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/DurationThresholdConfig.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * -\-\-
+ * Spotify Apollo Metrics Module
+ * --
+ * Copyright (C) 2013 - 2016 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.apollo.metrics.semantic;
+
+import com.google.common.collect.ImmutableMap;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigObject;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class DurationThresholdConfig {
+
+  private final Optional<Integer> allEndpoints;
+  private final Map<String, Map<String,
+      Integer>> goals;
+
+  private DurationThresholdConfig(final Optional<Integer> allEndpoints,
+                                  final Map<String, Map<String,
+                                      Integer>> goals) {
+    this.allEndpoints = allEndpoints;
+    this.goals = goals;
+  }
+
+  public Optional<Integer> getDurationThresholdForEndpoint(final String method,
```

Comment:
This might be cleaner if you flatten the nested map into an `Map<String, Optional<Integer>>` and use the endpoint as the key in the form you expect: `METHOD:URI`.

If you treat allEndpoints as the default value this could be simplified to something like this:
```
return thresholds.getOrDefault(endpoint, allEndpoints);
```

Does that make sense?